lib/benchmarks/gallery_recorder.dart

```diff
@@ -0,0 +1,48 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import 'package:gallery/benchmarks/gallery_automator.dart';
+
+import 'package:web_benchmarks/client.dart';
+
+/// A recorder that measures frame building durations for the Gallery.
+class GalleryRecorder extends WidgetRecorder {
+  GalleryRecorder({
+    @required this.benchmarkName,
+    this.shouldRunPredicate,
+    this.testScrollsOnly = false,
+  }) : assert(testScrollsOnly || shouldRunPredicate != null),
+        super(name: benchmarkName, useCustomWarmUp: true);
+
+  /// The name of the gallery benchmark to be run.
```

Comment:
mention that names are available in `common.dart`

Reply:
Added documentation.